src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java

```diff
@@ -212,6 +227,42 @@ private GapicSurfaceTestAssertView createAssertView(String expected, String actu
     return initFieldStructure;
   }
 
+  private Map<String, Object> createTestRequestInitFieldStructure(
+      MethodTransformerContext context, Iterable<Field> fields) {
+    ArrayList<String> fieldStrings = new ArrayList<>();
+    createTestRequestInitFieldStructure(context, fieldStrings, fields, "", 1);
+    return FieldStructureParser.parseFields(fieldStrings, createInitValueMap(context));
+  }
+
+  private void createTestRequestInitFieldStructure(
+      MethodTransformerContext context,
+      ArrayList<String> fieldStrings,
+      Iterable<Field> fields,
+      String prefix,
+      int depth) {
+    for (Field field : fields) {
+      String initString = prefix + field.getSimpleName();
+
+      if (field.getType().isMap()) {
+        // TODO(michaelbausor) support recursive map initialization
+        fieldStrings.add(initString);
+      } else {
+        if (field.getType().isRepeated()) {
+          initString += "[0]";
+        }
+        fieldStrings.add(initString);
+        if (field.getType().isMessage() && depth > 0) {
+          createTestRequestInitFieldStructure(
+              context,
+              fieldStrings,
+              field.getType().getMessageType().getFields(),
+              initString + ".",
+              depth - 1);
+        }
+      }
```

Comment:
I am going to think about how to improve the init sample code stuff simpler and easier to use for this case.